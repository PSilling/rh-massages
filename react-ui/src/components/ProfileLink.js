// react imports
import React, { Component } from 'react';

// util imports
import Auth from '../utils/Auth.js';
import _t from '../utils/Translations.js';

/**
 * Link that redirects to Keycloak account management.
 */
class ProfileLink extends Component {

  viewProfile = () => {
      Auth.keycloak.accountManagement();
    }

  render() {
    return(
      <button type="button" className="btn btn-link navbar-btn" onClick={() => this.viewProfile()}>
        <span className="glyphicon glyphicon-user"></span>&nbsp;
          { _t.translate('Profile') }
      </button>
    )
  }
}

export default ProfileLink
