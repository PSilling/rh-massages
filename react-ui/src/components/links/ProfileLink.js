// react imports
import React, { Component } from "react";

// module imports
import { NavLink, Tooltip } from "reactstrap";

// util imports
import Auth from "../../util/Auth";
import _t from "../../util/Translations";
import Util from "../../util/Util";

/**
 * Link that redirects to Keycloak account management.
 */
class ProfileLink extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  viewProfile = () => {
    Auth.keycloak.accountManagement();
  };

  render() {
    return (
      <span>
        <NavLink id={this.tooltipTarget} onClick={this.viewProfile} style={{ cursor: "pointer" }}>
          <span className="fas fa-user-circle" />
          &nbsp;
          {_t.translate("Profile")}
        </NavLink>
        <Tooltip isOpen={this.state.tooltipActive} target={this.tooltipTarget} toggle={this.toggleTooltip}>
          {_t.translate("Show Keycloak profile")}
        </Tooltip>
      </span>
    );
  }
}

export default ProfileLink;
