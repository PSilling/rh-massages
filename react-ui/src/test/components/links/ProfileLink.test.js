// react imports
import React from "react";
import TestRenderer from "react-test-renderer";

// test imports
import ProfileLink from "../../../components/links/ProfileLink";

test("renders content correctly", () => {
  const testRenderer = TestRenderer.create(<ProfileLink />);
  const testInstance = testRenderer.root;
  const button = testInstance.findByType("button");
  const treeJSON = testRenderer.toJSON();

  testInstance.findByProps({ className: "glyphicon glyphicon-user" });

  testInstance.instance.viewProfile = jest.fn();
  testRenderer.update(<ProfileLink />);
  button.props.onClick();

  expect(testInstance.instance.viewProfile).toHaveBeenCalledTimes(1);
  expect(treeJSON).toMatchSnapshot();
});
