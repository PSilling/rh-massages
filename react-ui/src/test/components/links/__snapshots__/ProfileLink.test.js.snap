// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileLink />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          id="Tooltip1"
          onClick={[Function]}
          style={
                    Object {
                              "cursor": "pointer",
                            }
          }
          tag="a"
>
          <span
                    className="fas fa-user-circle"
          />
           
          Profil
</NavLink>,
        <Tooltip
          autohide={true}
          delay={
                    Object {
                              "hide": 250,
                              "show": 0,
                            }
          }
          hideArrow={false}
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          target="Tooltip1"
          toggle={[Function]}
>
          Zobrazit Keycloak profil
</Tooltip>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span
              className="fas fa-user-circle"
/>,
            " ",
            "Profil",
          ],
          "id": "Tooltip1",
          "onClick": [Function],
          "style": Object {
            "cursor": "pointer",
          },
          "tag": "a",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-user-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          " ",
          "Profil",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autohide": true,
          "children": "Zobrazit Keycloak profil",
          "delay": Object {
            "hide": 250,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-tooltip",
          "target": "Tooltip1",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": "Zobrazit Keycloak profil",
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            id="Tooltip1"
            onClick={[Function]}
            style={
                        Object {
                                    "cursor": "pointer",
                                  }
            }
            tag="a"
>
            <span
                        className="fas fa-user-circle"
            />
             
            Profil
</NavLink>,
          <Tooltip
            autohide={true}
            delay={
                        Object {
                                    "hide": 250,
                                    "show": 0,
                                  }
            }
            hideArrow={false}
            isOpen={false}
            placement="top"
            placementPrefix="bs-tooltip"
            target="Tooltip1"
            toggle={[Function]}
>
            Zobrazit Keycloak profil
</Tooltip>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className="fas fa-user-circle"
/>,
              " ",
              "Profil",
            ],
            "id": "Tooltip1",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
            "tag": "a",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-user-circle",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " ",
            "Profil",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autohide": true,
            "children": "Zobrazit Keycloak profil",
            "delay": Object {
              "hide": 250,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-tooltip",
            "target": "Tooltip1",
            "toggle": [Function],
          },
          "ref": null,
          "rendered": "Zobrazit Keycloak profil",
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
